src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        contact = [select AccountID from Contact where ID = :contact.ID];
+        System.assertNotEquals(null, contact.AccountID);
+        
+        //Give address info to account
+        Account acc = [select BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry
+        from Account where ID = :contact.AccountID];
+        acc.BillingStreet = '123 main st';
+        acc.BillingState = 'IL';
+        acc.BillingPostalCode = '54545';
+        acc.BillingCountry = 'USA';
+
+        Test.startTest();
+        update acc;
+        Test.stopTest();
+        
+        //Verify child address record was created
+        List<Address__c> addrs = [select Parent_Contact__c from Address__c where Parent_Account__c = :acc.ID]; 
+        System.assertEquals(1, addrs.size());
+        System.assertEquals(null, addrs[0].Parent_Contact__c);
+        
+         //Verify child contact has no address info
+        contact = [select MailingStreet, MailingCity, MailingState, MailingPostalCode, MailingCountry
+        from Contact where ID = :contact.ID];
+        System.assertEquals(null, contact.MailingStreet);
+        System.assertEquals(null, contact.MailingState);
+        System.assertEquals(null, contact.MailingPostalCode);
+        System.assertEquals(null, contact.MailingCountry);
+        
+        //Verify no address record was created as child of the child contact
+        addrs = [select ID from Address__c where Parent_Contact__c = :contact.ID];
+        System.assertEquals(0, addrs.size()); 
+    }
+  
+    /*********************************************************************************************************
+    @description Create N new Contacts with Mailing addresses 
+    verify:
+        N addresses created
+        Contact address matches address object address
+    **********************************************************************************************************/            
+    @isTest
+    public static void newRecordsWithAddrInfo() {            
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+ 
+        Integer contactCount = 3;
+        List<Contact> contacts = UTIL_UnitTestData_TEST.createMultipleTestContacts(contactCount);
+        for (Contact contact : contacts) {
+            contact.MailingStreet = '123 45th';
+            contact.MailingCity = 'Seattle';
+        }
+        Test.startTest();
+        insert contacts;
+        Test.stopTest();
+
+        // verify results
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        system.assertEquals(contactCount, mapAccIdAcc.size());
+        
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, Parent_Contact__c, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.